src/places/places.service.ts

```diff
@@ -38,77 +44,92 @@ export class PlacesService extends createPrismaBase(MODELS.Place) {
-    )[]
+    const placeSelectBase: Prisma.PlaceSelect | undefined = placeColumns
+      ? (buildColumnSelect(placeColumns) as Prisma.PlaceSelect)
+      : undefined
 
     const buildRaceInclude = () => {
-      if (raceColumns) {
-        const raceSelect = buildColumnSelect(raceColumns) as Prisma.RaceSelect
-        return { select: raceSelect }
-      } else {
-        return true
+      if (!raceColumns) return true
+      if (!includeRaces) return true
+
+      // Force add slug and positionNames so we can dedupe by slug
+      const cols = Array.from(
+        new Set([
+          ...raceColumns.split(','),
+          SLUG_COLUMN_NAME,
+          POSITION_NAMES_COLUMN_NAME,
+        ]),
+      ).join(',')
+
+      return {
+        select: buildColumnSelect(cols) as Prisma.RaceSelect,
       }
     }
 
-    if (!includeChildren && !includeParent) {
-      // Just get one level of place(s)
-      if (placeColumns) {
-        const select = buildColumnSelect(placeColumns) as Prisma.PlaceSelect
-
-        if (includeRaces) {
-          select.Races = buildRaceInclude()
-        }
+    const makePlaceSelection = (withRaces: boolean) => {
+      if (!placeSelectBase) {
+        if (!withRaces) return true
 
-        places = await this.model.findMany({ where, select })
-      } else {
-        const include: Prisma.PlaceInclude = {}
-        if (includeRaces) {
-          include.Races = buildRaceInclude()
+        return {
+          include: {
+            Races: buildRaceInclude(),
+          },
         }
-        places = await this.model.findMany({ where, include })
       }
-    } else {
-      // Get multiple levels of places
-      const include: Prisma.PlaceInclude = {}
+      const sel: Prisma.PlaceSelect = { ...placeSelectBase }
+      if (withRaces) sel.Races = buildRaceInclude()
+      return { select: sel }
+    }
 
-      if (includeChildren) {
-        include.children = includeRaces
-          ? { include: { Races: buildRaceInclude() } }
-          : true
-      }
+    let queryArgs: Prisma.PlaceFindManyArgs
 
-      if (includeParent) {
-        include.parent = includeRaces
-          ? { include: { Races: buildRaceInclude() } }
-          : true
+    if (placeSelectBase) {
+      const rootSelect: Prisma.PlaceSelect = {
+        ...placeSelectBase,
+        ...(includeRaces && { Races: buildRaceInclude() }),
+        ...(includeChildren && { children: makePlaceSelection(includeRaces) }),
+        ...(includeParent && { parent: makePlaceSelection(includeRaces) }),
       }
 
-      if (includeRaces) {
-        include.Races = buildRaceInclude()
+      queryArgs = { where, select: rootSelect }
+    } else {
+      queryArgs = {
+        where,
+        include: {
+          ...(includeRaces && { Races: buildRaceInclude() }),
+          ...(includeChildren && {
+            children: makePlaceSelection(includeRaces),
+          }),
+          ...(includeParent && { parent: makePlaceSelection(includeRaces) }),
```

Comment:
If you find yourself calling the same function over and over again with the same values, returning the same result, in the same scope... that's gonna be a really big indicator that the results should be stored either in that scope or in a hire scope, and just referenced from their stored value.